### ISSUE TITLE
paliaOnMac Not detecting Whisky Bottle

### ISSUE THREAD

**Issue by kpanattoni** [DanMossa/paliaOnMac-release#22]
Hello there,

When trying to install, the app does not detect the Whisky bottle at all. I have gone so far as to completely delete Whisky including any preference, cache or container files I can find, so that I can install a fresh version. I downloaded the newest version of paliaOnMac and upon launch I get the screen telling me how to set up Whisky. I follow the steps in the dialogue, the bottle stops spinning as it says (note: there is no "refresh" button on the lower right), close Whisky as it says, click done, and paliaOnMac will still not detect a bottle. I deleted everything related to Whisky again and chose to allow paliaOnMac download Whisky for me and went through the steps of downloading it and installing it fresh, then setting up a bottle precisely as it asked and named it Palia. Clicking done in the dialogue box, closed Whisky and it still did not find a bottle. I closed both paliaOnMac and Whisky and ran paliaOnMac again and still not finding a bottle. I have downloaded the latest versions of both apps going through this process over and over. Palia itself does work in Crossover, but I would like to see an improvement if this app will help. System info and app versions below.

 Mac Mini M1 2020
 MacOS Sonoma macOS 14.4.1 (23E224)
  Model Name:	Mac mini
  Model Identifier:	Macmini9,1
  Model Number:	MGNR3LL/A
  Chip:	Apple M1
  Total Number of Cores:	8 (4 performance and 4 efficiency)
  Memory:	8 GB
  System Firmware Version:	10151.101.3
  OS Loader Version:	10151.101.3
  Hardware UUID:	EC2BEE4C-C249-570F-97F4-C96FC6B811C4
  Provisioning UDID:	00008103-0010083A2131001E
  Activation Lock Status:	Enabled

Software versions attempted:
PaliaOnMac: 0.7.1, 0.6.2
Whisky: 2.2.3, 2.2.4

![Screenshot 2024-04-08 at 9 39 39 PM](https://github.com/DanMossa/paliaOnMac-release/assets/85253157/c9f02fe8-265e-4a56-937d-299e6b78b8b5)

**Comment by DanMossa** [comment]
Looking into it!

**Comment by kpanattoni** [comment]
@DanMossa 

I do not have a subfolder for "Bottles". These are the contents of the  ~/Library/Containers/com.isaacmarovitz.Whisky/ folder:

B3CB61BE-9C14-41C3-90AB-1CD1A0369184	BottleVM.plist

**Comment by kpanattoni** [comment]
@DanMossa 

It is worth noting that I do not have enough internal drive space to keep bottles, I have to store them on an external SSD.

**Comment by kpanattoni** [comment]
@DanMossa 

Here is a cat of BottleVM.plist, if it helps any:

MyKnee:com.isaacmarovitz.Whisky kim$ cat BottleVM.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>fileVersion</key>
	<dict>
		<key>build</key>
		<string></string>
		<key>major</key>
		<integer>1</integer>
		<key>minor</key>
		<integer>0</integer>
		<key>patch</key>
		<integer>0</integer>
		<key>preRelease</key>
		<string></string>
	</dict>
	<key>paths</key>
	<array>
		<dict>
			<key>relative</key>
			<string>file:///Volumes/SSD/Users/Shared/Whiskey/Palia/</string>
		</dict>
	</array>
</dict>
</plist>

**Comment by kpanattoni** [comment]
Firstly, one thing to note: Whenever I go to look at this directory in the Finder, it renames the folder to simply 'Whisky' and does not show it beginning with 'com.' I had to pull some ridiculousness with both the finder and the terminal to get this to happen.Through a lot of jim-jangling around with Whisky and the terminal, this is what I was able to produce. Note the directory structure:


MyKnee:Containers kim$ cat ~/Library/Containers/com.isaacmarovitz.Whisky/Palia/Metadata.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>dxvkConfig</key>
	<dict>
		<key>dxvk</key>
		<false/>
		<key>dxvkAsync</key>
		<true/>
		<key>dxvkHud</key>
		<dict>
			<key>off</key>
			<dict/>
		</dict>
	</dict>
	<key>fileVersion</key>
	<dict>
		<key>build</key>
		<string></string>
		<key>major</key>
		<integer>1</integer>
		<key>minor</key>
		<integer>0</integer>
		<key>patch</key>
		<integer>0</integer>
		<key>preRelease</key>
		<string></string>
	</dict>
	<key>info</key>
	<dict>
		<key>blocklist</key>
		<array/>
		<key>name</key>
		<string>Palia</string>
		<key>pins</key>
		<array/>
	</dict>
	<key>metalConfig</key>
	<dict>
		<key>metalHud</key>
		<false/>
		<key>metalTrace</key>
		<false/>
	</dict>
	<key>wineConfig</key>
	<dict>
		<key>enhancedSync</key>
		<dict>
			<key>msync</key>
			<dict/>
		</dict>
		<key>windowsVersion</key>
		<string>win10</string>
		<key>wineVersion</key>
		<dict>
			<key>build</key>
			<string></string>
			<key>major</key>
			<integer>7</integer>
			<key>minor</key>
			<integer>7</integer>
			<key>patch</key>
			<integer>0</integer>
			<key>preRelease</key>
			<string></string>
		</dict>
	</dict>
</dict>
</plist>

**Comment by kpanattoni** [comment]
And note: Palia On Mac still says there is no bottle and asks if I want to set up Whisky.

**Comment by kpanattoni** [comment]
The path in terminal is correct, and the folder exists in the GUI reads as Whisky but when you get info on it, the proper structure is there. I have tried to force the GUI to read the same way as its path and it always changes it back to simply "Whisky". Attached is a screenshot of this phenomenon.
![Screenshot 2024-04-23 at 5 43 15 AM](https://github.com/DanMossa/paliaOnMac-release/assets/85253157/d5a5c717-ad3c-4e10-ac3a-fdc54b793fd9)

**Comment by kpanattoni** [comment]
Thoughts?

**Comment by DanMossa** [comment]
Hey! Yeah, the issue is that the Palia on Mac installer requires everything on 1 drive

**Comment by kpanattoni** [comment]
1 drive? Do you mean on the main drive, which in my case would be Macintosh HD? I tried that once more and I receive the same errors, please see screenshots attached. This process loops over and over again. When it says to close Whisky, I close it, then click on "Done" and am then greeted by "Didn't find Whisky with a Bottle." and the whole thing starts over again. This is not working. How do I fix it? You can see the path says it is on my internal drive... I even tried to rename my folder "Palia" and then Whisky does not recognise it. I deleted all of my bottles from Whisky but then imported the bottle named Palia from the path ~/Library/Containers/Whisky/Palia and Whisky recognised it, but PaliaOnMac did not. What is happening here?

![Screenshot 2024-07-27 at 9 11 36 PM](https://github.com/user-attachments/assets/da299480-a56f-4c4c-8355-3e84ec2d7eb4)

![Screenshot 2024-07-27 at 9 01 30 PM](https://github.com/user-attachments/assets/66517dce-1388-41ff-8a38-b4599cb7ca7f)

![Screenshot 2024-07-27 at 8 59 59 PM](https://github.com/user-attachments/assets/a18bfb95-7de4-4457-ae34-64d603fe8225)
![Screenshot 2024-07-27 at 9 00 59 PM](https://github.com/user-attachments/assets/3581b484-b352-4cb9-9c12-ad50bd4366b9)

![Screenshot 2024-07-27 at 8 57 14 PM](https://github.com/user-attachments/assets/71b1e3da-89fa-4daf-857f-084dd99c4ab0)

**Comment by DanMossa** [comment]
My suggestion is to completely nuke Whisky and all created folders from Whisky, and then try again using this launcher. You can join the Discord via the link below for more real time help also!

**Comment by kpanattoni** [comment]
> My suggestion is to completely nuke Whisky and all created folders from Whisky, and then try again using this launcher. You can join the Discord via the link below for more real time help also!

I have a feeling it will need to be Discord. I have described above that I have already done what you suggested by deleting everything Whisky related on my computer. I even unmounted any of my external hard drives that had my Whisky downloads saved on it or even the word Whisky in folders or anything else. I detailed that in my posts above, or at least I hope I did. In any case, there is nothing at all related to Whisky in my system, and I even did this at the CLI level by searching through the entire OS with find and grep tools as root to prevent any permissions related issues in my search. The problem has persisted. Either there is something I am severely missing here, or we need to have a conversation in more detail. :P 

Thanks.
